src/main/java/org/testng/xml/XmlTest.java

```diff
@@ -752,4 +752,17 @@ public void setGroups(XmlGroups xmlGroups) {
   public XmlGroups getXmlGroups() {
     return m_xmlGroups;
   }
+
+  /**
+   * @param names The list of names to check.
+   * @return <code>true</code> if the current test's name matches with any of the given names.
+   */
+  public boolean nameMatchesAny(List<String> names) {
```

Comment:
I think for this case, we can skip having to create a separate utility class because this is effectively speaking not a business logic, but just something that works with an internal attribute. So I am going to leave this as is.

Reply:
XmlObjects should just be POJO for the xml binding without business logic, could you move this method in an external util class?